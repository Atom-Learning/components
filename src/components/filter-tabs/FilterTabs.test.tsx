import { render, screen, fireEvent } from '@testing-library/react'
import React from 'react'
import { axe } from 'jest-axe'

import { FilterTabs } from './'

const filters = ['tab1', 'tab2', 'tab3']

describe('FilterTabs component', () => {
  it('renders', async () => {
    const { container } = await render(
      <FilterTabs filters={filters} value="tab1" onValueChange={() => null} />
    )

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <FilterTabs filters={filters} value="tab1" onValueChange={() => null} />
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('changes tab when items are clicked on', async () => {
    let currentTab = 'tab1'

    render(
      <FilterTabs
        filters={filters}
        value="tab1"
        onValueChange={(value) => (currentTab = value)}
      />
    )

    const tab2 = await screen.findByText('tab2')
    fireEvent.click(tab2)

    expect(currentTab).toBe('tab2')
  })

  it("doesn't allow clicking on disabled items", async () => {
    let currentTab = 'tab1'

    render(
      <FilterTabs
        filters={filters}
        value="tab1"
        onValueChange={(value) => (currentTab = value)}
        disabled
      />
    )

    const tab2 = await screen.findByText('tab2')
    fireEvent.click(tab2)

    expect(currentTab).toBe('tab1')
  })
})
