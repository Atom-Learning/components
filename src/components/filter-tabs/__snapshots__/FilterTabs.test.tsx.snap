// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTabs component renders 1`] = `
@media  {
  .c-cmpvrW {
    position: relative;
  }

  .c-dhzjXW {
    display: flex;
  }

  .c-bTGMgd {
    flex-shrink: 0;
    border-bottom: 1px solid var(--colors-tonal300);
    overflow-x: auto;
    width: 100%;
    cursor: pointer;
  }

  .c-eeDzjx {
    font-family: var(--fonts-body);
    padding: var(--space-4);
    height: var(--sizes-5);
    flex-shrink: 0;
  }

  .c-eeDzjx:hover {
    background: rgba(30,113,246,0.1);
  }

  .c-eeDzjx[data-disabled=true] {
    background: unset;
    color: var(--colors-tonal300);
    cursor: not-allowed;
  }
}

@media  {
  .c-eeDzjx-jJnYiF-selected-true {
    font-weight: 600;
    color: var(--colors-primary);
  }
}

<div>
  <div
    class="c-cmpvrW"
  >
    <div
      class="c-dhzjXW c-bTGMgd"
    >
      <div
        class="c-dhzjXW c-eeDzjx c-eeDzjx-jJnYiF-selected-true"
        data-disabled="false"
      >
        tab1
      </div>
      <div
        class="c-dhzjXW c-eeDzjx"
        data-disabled="false"
      >
        tab2
      </div>
      <div
        class="c-dhzjXW c-eeDzjx"
        data-disabled="false"
      >
        tab3
      </div>
    </div>
  </div>
</div>
`;
